src/components/DistanceMapView/index.android.js

```diff
@@ -0,0 +1,48 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import _ from 'underscore';
+import BlockingView from '../BlockingViews/BlockingView';
+import MapView from '../MapView';
+import styles from '../../styles/styles';
+import useNetwork from '../../hooks/useNetwork';
+import useLocalize from '../../hooks/useLocalize';
+import * as Expensicons from '../Icon/Expensicons';
+import * as StyleUtils from '../../styles/StyleUtils';
+import * as distanceMapViewPropTypes from './distanceMapViewPropTypes';
+
+function DistanceMapView(props) {
+    const [isMapReady, setIsMapReady] = useState(false);
+    const {isOffline} = useNetwork();
+    const {translate} = useLocalize();
+
+    return (
+        <>
+            <MapView
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {..._.omit(props, 'overlayStyle')}
+                onMapReady={() => {
+                    if (isMapReady) {
+                        return;
+                    }
+                    setIsMapReady(true);
+                }}
+            />
+            {!isMapReady && (
+                <View style={StyleUtils.combineStyles(styles.mapViewOverlay, props.overlayStyle)}>
+                    <BlockingView
+                        icon={Expensicons.EmptyStateRoutePending}
+                        title={translate('distance.mapPending.title')}
+                        subtitle={isOffline ? translate('distance.mapPending.subtitle') : translate('distance.mapPending.onlineSubtitle')}
+                        shouldShowLink={false}
+                    />
+                </View>
+            )}
```

Comment:
This code block introduced the following issue:
- https://github.com/Expensify/App/issues/43420

Which was fixed by showing `BlockingView` **only** when `!requireRouteToDisplayMap`, otherwise show `PendingMapView`.

> /* The "map pending" text should only be shown in the IOU create flow. In the created IOU preview, only the icon should be shown. */